src/stores/search.ts

```diff
@@ -29,313 +30,282 @@ import {
-    const filterEntries = Object.entries(searchFilters.value) as [
-      string,
-      FilterItem[]
-    ][]
-    return filterEntries.some(
-      ([filterKey, filterItems]) =>
-        filterKey !== 'mature' && filterItems.some((filter) => filter.checked)
-    )
-  })
-
-  // Actions
-  /**
-   * After a search type is changed, unchecks all the filters that are not
-   * applicable for this Media type.
-   */
-  function clearOtherMediaTypeFilters(searchType: SupportedSearchType) {
-    const mediaTypesToClear = supportedSearchTypes.filter(
-      (type) => type !== searchType
-    )
-    const filterKeysToClear = mediaTypesToClear.reduce((acc, type) => {
-      return [...acc, ...mediaUniqueFilterKeys[type]]
-    }, [] as FilterCategory[])
-
-    allFilterCategories.forEach((filterCategory) => {
-      if (filterKeysToClear.includes(filterCategory)) {
-        state.filters[filterCategory] = state.filters[filterCategory].map(
-          (f) => ({
-            ...f,
-            checked: false,
-          })
+    },
+    /**
+     * Returns the number of checked filters, excluding the `mature` filter.
+     */
+    appliedFilterCount(state) {
+      const filterKeys = mediaFilterKeys[state.searchType].filter(
+        (f) => f !== 'mature'
+      )
+      return filterKeys.reduce((count, filterCategory) => {
+        return (
+          count + state.filters[filterCategory].filter((f) => f.checked).length
         )
-      }
-    })
-  }
-
-  /**
-   * Replaces filters with the newFilterData object that was created using initial filters,
-   * and setting parameters from the search query to checked.
-   *
-   *
-   *
-   */
-  function replaceFilters(newFilterData: Filters) {
-    allFilterCategories.forEach((filterCategory) => {
-      state.filters[filterCategory] = newFilterData[filterCategory]
-    })
-  }
-
-  /**
-   * Merge providers from API response with the filters that came from the browse URL search query string
-   * and match the checked properties in the store.
-   */
-  function initProviderFilters({
-    mediaType,
-    providers,
-  }: {
-    mediaType: SupportedMediaType
-    providers: { source_name: string; display_name: string }[]
-  }) {
-    const providersKey: FilterCategory = `${mediaType}Providers`
-    const currentProviders = state.filters[providersKey]
-      ? [...state.filters[providersKey]]
-      : []
-    state.filters[providersKey] = providers.map((provider) => {
-      const existingProviderFilterIdx = currentProviders.findIndex(
-        (p) => p.code === provider.source_name
+      }, 0)
+    },
+    /**
+     * Returns the object with filters for selected search type, with codes, names for i18n labels, and checked status.
+     */
+    searchFilters(state) {
+      return mediaFilterKeys[state.searchType]
+        .filter((filterKey) => filterKey !== 'mature')
+        .reduce((obj, filterKey) => {
+          obj[filterKey] = this.filters[filterKey]
```

Comment:
Should we choose to use either `this` or the explicit `state` parameter consistently? Mixing the two is confusing.

Reply:
Ah interesting. I guess it's fine to mix them, I will get used to it!